packages/devtools_app/lib/src/logging/logging_controller.dart

```diff
@@ -635,6 +609,58 @@ class LoggingController {
       ),
     );
   }
+
+  @override
+  List<LogData> matchesForSearch(String search) {
+    if (search == null || search.isEmpty) return [];
+    final matches = <LogData>[];
+    final caseInsensitiveSearch = search.toLowerCase();
+
+    final currentLogs = filteredData.value;
+    for (final log in currentLogs) {
+      if ((log.summary != null &&
+              log.summary.toLowerCase().contains(caseInsensitiveSearch)) ||
+          (log.details != null &&
+              log.details.toLowerCase().contains(caseInsensitiveSearch))) {
+        matches.add(log);
+      }
+    }
+    return matches;
+  }
+
+  @override
+  void filterData(QueryFilter filter) {
+    if (filter == null) {
+      filteredData.value = List.from(data);
+    } else {
+      filteredData.value = data.where((log) {
+        for (final arg in filter.filterArguments) {
+          if (arg.id == kindFilterId) {
+            if (!arg.matchesValue(log.kind.toLowerCase())) {
+              return false;
+            }
+          }
+        }
+        if (filter.substrings.isNotEmpty) {
+          for (final substring in filter.substrings) {
+            final caseInsensitiveSubstring = substring.toLowerCase();
+            final matchesKind = log.kind != null &&
+                log.kind.toLowerCase().contains(caseInsensitiveSubstring);
+            final matchesSummary = log.summary != null &&
+                log.summary.toLowerCase().contains(caseInsensitiveSubstring);
+            final matchesDetails = log.details != null &&
+                log.summary.toLowerCase().contains(caseInsensitiveSubstring);
+            if (matchesKind || matchesSummary || matchesDetails) {
```

Comment:
nit: why not just return immediately at the point each of these is true rather than computing them all and then returning?